tez-dag/src/test/java/org/apache/tez/dag/app/TestDAGAppMaster.java

```diff
@@ -332,6 +339,92 @@ public void testParseAllPluginsCustomAndYarnSpecified() throws IOException {
     assertEquals(TC_NAME + CLASS_SUFFIX, tcDescriptors.get(1).getClassName());
   }
 
+  @Test
+  public void testShutdownTezAMWithMissingRecovery() throws Exception {
+
+    TezConfiguration conf = new TezConfiguration();
+    conf.setBoolean(TezConfiguration.TEZ_AM_CREDENTIALS_MERGE, true);
+    conf.setBoolean(TezConfiguration.TEZ_LOCAL_MODE, true);
+    conf.set(TezConfiguration.TEZ_AM_STAGING_DIR, TEST_DIR.toString());
+    conf.setBoolean(TezConfiguration.TEZ_AM_FAILURE_ON_MISSING_RECOVERY_DATA, true);
+    conf.setBoolean(TezConfiguration.DAG_RECOVERY_ENABLED, true);
+    ApplicationId appId = ApplicationId.newInstance(1, 1);
+    ApplicationAttemptId attemptId = ApplicationAttemptId.newInstance(appId, 2);
+
+    FileSystem spyRecoveryFs = spy(new FileSystem() {
```

Comment:
is this spy reusable in other testing methods in this class? if so, refactor it to class field, if it's not, make it obvious why is this special (with variable name and/or comment on implemented methods)

Reply:
made it common for the class